cpp/include/cudf/io/text/detail/multistate.hpp

```diff
@@ -37,7 +37,7 @@ struct multistate {
      * @brief Creates a segment which represents (0, 0]
      */
 
-    constexpr multistate_segment() : _data(0) {}
+    constexpr multistate_segment() {}
```

Comment:
Shouldn't this be explicitly defaulted instead of empty? 
```suggestion
    constexpr multistate_segment() = default;
```

Reply:
Currently, I have the `= default` modernize check turned off because the autofix is broken. I can this change locally tho.